vcd/datasource_vcd_resource_list.go

```diff
@@ -1110,6 +1110,56 @@ func vdcTemplateList(d *schema.ResourceData, meta interface{}) (list []string, e
 	return genericResourceList(d, "vcd_org_vdc_template", ancestors, items)
 }
 
+func nsxtAlbServiceEngineGroup(d *schema.ResourceData, meta interface{}) (list []string, err error) {
+	client := meta.(*VCDClient)
+	allSegs, err := client.GetAllAlbServiceEngineGroups("", nil)
+	if err != nil {
+		return nil, err
+	}
+	items := make([]resourceRef, len(allSegs))
+	for i, seg := range allSegs {
+		items[i] = resourceRef{
+			name:   seg.NsxtAlbServiceEngineGroup.Name,
+			id:     seg.NsxtAlbServiceEngineGroup.ID,
+			href:   fmt.Sprintf("%s/cloudapi/%s%s%s", client.Client.VCDHREF.String(), types.OpenApiPathVersion1_0_0, types.OpenApiEndpointAlbServiceEngineGroups, seg.NsxtAlbServiceEngineGroup.ID),
+			parent: "System",
+		}
+	}
+	return genericResourceList(d, "vcd_nsxt_alb_service_engine_group", nil, items)
+}
+
+func nsxtAlbServiceEngineGroupAssignment(d *schema.ResourceData, meta interface{}) (list []string, err error) {
+	client := meta.(*VCDClient)
+
+	orgName, vdcName, _, _, egw, err := getEdgeGatewayDetails(d, meta)
+	if err != nil {
+		return list, fmt.Errorf("error retrieving edge gateway '%s': %s ", d.Get("parent").(string), err)
+	}
```

Comment:
There is a bit of a problem that this function uses "NSX-V" Edge Gateway function which can cause some unexpected error. 
IMO would be good to validate that this is an NSX-T Edge Gateway (we don't have ALB for NSX-V). Either on the edge gateway, or even checking if the VDC is an NSX-T one.

Both `AnyTypeEdgeGateway` and `Vdc` have `IsNsxt` methods

Reply:
Good point, I tried to reuse this function to avoid adding more code but turned out to be a bad idea 😄 
Now it should be good, it checks NSX-T and uses NSX-T Gateway types/methods.